app/presenters/migration/new.rb

```diff
@@ -15,6 +15,112 @@ def page_content_heading
       'Tell your story'
     end
 
+    def fields
```

Comment:
Could you extend the [generic field template](https://github.com/europeana/Europeana-Patternlab/blob/develop/source/_patterns/molecules/forms/generic-field.mustache) to also accept:
1. an `error` key for each field, for error messages from server-side validation, displayed with some prominence
2. a `hint` key for each fields, for a longer description of the field which may or may not be present, but if it is would be displayed beneath the field in smaller text

Reply:
I've added these.  Also added are:
* data_url
* data_param

If "data_url" is set to "genonames" and "data_param" is set to "name" then after the user types (2 chars) into the field the page will make a request to:

  localhost:3000/geonames?name=ab

The expected return is an array of strings - the front end will handle the highlighting of the typed term.

Reply:
We need to handle different text and values in the autocomplete. For vocabularies like geonames.org, we want to display to the user a readable label, but submit to the app, and store, the URI. Conversely, we need a way to re-display the label to the user when the form field comes pre-populated with a URI, by immediately performing a different lookup query to the API in question.

I also need to be able to specify the attributes for the <form> element. Specifically: enctype, action, accept-charset and method. But ideally any attribute should be settable by back-end without adjustment to the template.